File: repos_estudiantes_TARQUI_primera_entrega/REQUENA_OROS_ADRIAN_MARCELO/herencia/Herencia/src/ejercicio3/Animal.java
package ejercicio3;

public class Animal {
    private String nombre;
    private int edad;

    public Animal(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
    }

    public void desplazarse() {
        System.out.println("El animal se está desplazando");
    }
    
    // Getters
    public String getNombre() { return nombre; }
    public int getEdad() { return edad; }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CARRERA_POZO_ALEJANDRA/Herencia/Ejercicio3/Animal.py
# Clase base
class Animal:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

    def desplazarse(self):
        print(f"{self.nombre} se desplaza de alguna manera.")
class Leon(Animal):
    def desplazarse(self):
        print(f"{self.nombre} camina con fuerza por la sabana.")
class Pinguino(Animal):
    def desplazarse(self):
        print(f"{self.nombre} nada y se desliza por el hielo.")
class Canguro(Animal):
    def desplazarse(self):
        print(f"{self.nombre} salta ágilmente por el terreno.")
animales = [
    Leon("Simba", 5),
    Pinguino("Pingu", 3),
    Canguro("Jack", 6)]
for animal in animales:
    animal.desplazarse()
